apis/v1alpha2/policy_types.go

```diff
@@ -31,12 +31,29 @@ const (
 	PolicyLabelKey = "gateway.networking.k8s.io/policy"
 )
 
-// PolicyTargetReference identifies an API object to apply a direct or
+// LocalPolicyTargetReference identifies an API object to apply a direct or
 // inherited policy to. This should be used as part of Policy resources
 // that can target Gateway API resources. For more information on how this
 // policy attachment model works, and a sample Policy resource, refer to
 // the policy attachment documentation for Gateway API.
-type PolicyTargetReference struct {
+type LocalPolicyTargetReference struct {
+	// Group is the group of the target resource.
+	Group Group `json:"group"`
+
+	// Kind is kind of the target resource.
+	Kind Kind `json:"kind"`
+
+	// Name is the name of the target resource.
+	Name ObjectName `json:"name"`
+}
+
+// NamespacedPolicyTargetReference identifies an API object to apply a direct or
+// inherited policy to, potentially in a different namespace. This should only
+// be used as part of Policy resources that need to be able to target resources
+// in different namespaces. For more information on how this policy attachment
+// model works, and a sample Policy resource, refer to the policy attachment
+// documentation for Gateway API.
+type NamespacedPolicyTargetReference struct {
```

Comment:
I'm not sure but it looks like this type may be missing from some places where `PolicyTargetReference` has been swapped to `LocalPolicyTargetReference`?

Reply:
Good catch - that was an intentional omission for now, basically leaving out `NamespacedPolicyTargetReferenceWithSectionName` until we're very sure we need it. It should be easy for us to add in the future or for someone else to recreate on their own if needed, just generally think that namespaced targets in policies should be a rare exception. Happy to change though if you'd prefer I add all the types up front.

Reply:
Make sense, agreed keeping this simpler unless/until we have an actual need is best!